### ISSUE TITLE
Questions regarding evaluation

### ISSUE THREAD

**Issue by bhsimon0810** [Clin0212/HydraLoRA#35]
Hi, thanks for your excellent work. I am trying to fine-tune the model using the example data, and encountering some problems during the  evaluation. I would be very grateful if you could provide some clarifications.

Following your document, I have made the following modifications:
 - Modified  `opencompass/opencompass/models/huggingface.py` (`xxxx` is the user name, please ignore):
```
import sys
sys.path.insert(0, '/home/xxxx/HydraLoRA/HydraLoRA')
```

- Modified `opencompass/configs/models/hf_llama/hf_llama2_7b.py`:

```
models = [
    dict(
        type=HuggingFaceCausalLM,
        abbr='llama-2-7b-hf',
        path="/home/xxxx/.cache/huggingface/hub/models--meta-llama--Llama-2-7b-hf/snapshots/01c7f73d771dfac7d292323805ebc428287df4f9",
        tokenizer="/home/xxxx/.cache/huggingface/hub/models--meta-llama--Llama-2-7b-hf/snapshots/01c7f73d771dfac7d292323805ebc428287df4f9",
        peft_path='/home/xxxx/HydraLoRA/HydraLoRA',
        max_out_len=1024,
        batch_size=8,
        run_cfg=dict(num_gpus=1),
    )
]
```

To evaluate the performance on `mmlu` dataset, I used the following command:
```
opencompass --models hf_llama2_7b --datasets mmlu_all_sets
```

But it turns out that the evaluation was not completed as expected, as you can see in the attached screenshot.

<img width="1263" alt="Image" src="https://github.com/user-attachments/assets/3ad30dd3-9cd3-4398-9f61-4644a1c71ab5" />

Could you kindly provide more details regarding the evaluation steps, such as the version of opencompass, the scripts/commands to run the evaluation? Your guidance would be highly appreciated. Thanks for your time and assistance!

**Comment by Ethan16162** [comment]
@bhsimon0810  Have you solved this problem? I have encountered the same issue when evaluate on `mmlu_all_sets`. However, my configuration in `opencompass/configs/models/hf_llama/hf_llama2_7b.py` is as follows:
``` 
models = [
    dict(
        type=HuggingFacewithChatTemplate,
        abbr='llama-3_2-3b-instruct-hf',
        path='/home/xxx/code/HydraLoRA/ckpt/llama-3.2-3B-Instruct',
        tokenizer_path='/home/xxx/code/HydraLoRA/ckpt/llama-3.2-3B-Instruct',
        peft_path='/home/xxx/code/HydraLoRA/output/lora_model/sft_lora_model',
        max_out_len=1024,
        batch_size=8,
        run_cfg=dict(num_gpus=1),
        stop_words=['<|end_of_text|>', '<|eot_id|>'],
        model_kwargs=dict(torch_dtype=torch.bfloat16),
    )
]
```

**Comment by bhsimon0810** [comment]
@YansongGuo1616 Hi, could you please clarify which scripts are used to evaluate `mmlu_all_sets`? Do you also use `opencompass --models your_model_name --datasets mmlu_all_sets` command? Additionally, it seems that the parameter `tokenizer_path` should be the same as `peft_path`, which in your case appears to be `/home/gys/code/HydraLoRA/output/lora_model/sft_lora_model`. 

I am now able to do the evaluation, but it seems there are still some problems with reproducing the results. I am still working on it. Please let me know if you'd like to discuss more.

**Comment by Ethan16162** [comment]
@bhsimon0810 Hello, of course! Below is my run and evaluation script：

```
PYTHONPATH="/home/xxx/code/HydraLoRA/HydraLoRA:$PYTHONPATH" opencompass --models hf_llama3_2_3b_instruct --datasets mmlu_gen --debug --hf-num-gpus 4
```

```
PYTHONPATH="/home/xxx/code/HydraLoRA/HydraLoRA:$PYTHONPATH" opencompass --models hf_llama3_2_3b_instruct --datasets gsm8k_gen --debug --hf-num-gpus 4
```
I am using the `hf_llama3_2_3b_instruct ` model, but the evaluation results are not very good at the moment.

**Comment by Ethan16162** [comment]
> [@YansongGuo1616](https://github.com/YansongGuo1616) Hi, could you please clarify which scripts are used to evaluate `mmlu_all_sets`? Do you also use `opencompass --models your_model_name --datasets mmlu_all_sets` command? Additionally, it seems that the parameter `tokenizer_path` should be the same as `peft_path`, which in your case appears to be `/home/gys/code/HydraLoRA/output/lora_model/sft_lora_model`.
> 
> I am now able to do the evaluation, but it seems there are still some problems with reproducing the results. I am still working on it. Please let me know if you'd like to discuss more.

By this peft path(`/home/gys/code/HydraLoRA/output/lora_model/sft_lora_model`), the evaluation could be conducted successfully, however, the reproduced accuracy is slightly lower than that of the untuned LLM. I'm not sure if it's due to path issues, training dataset problems, or fine-tuning issues.

**Comment by bhsimon0810** [comment]
@YansongGuo1616 Hi, I use almost the same evaluation scripts, but I have also noticed some variation in the results. I plan to downgrade the version of opencompass and re-run the experiments. I will let you know the result later. 

By the way, I am using the databricks-dolly-15k dataset, as mentioned in [issue 1](https://github.com/Clin0212/HydraLoRA/issues/1).

**Comment by bhsimon0810** [comment]
@YansongGuo1616 Hi, I used the following configuration to finetune the model:

```bash
lr=0.0002
lora_rank=8
lora_alpha=32
lora_trainable="gate_proj,down_proj,up_proj"
lora_dropout=0.05                 
pretrained_model=meta-llama/Llama-2-7b-hf
tokenizer_path=meta-llama/Llama-2-7b-hf
dataset_dir=data/databricks/train  # databricks-dolly-15k dataset  
validation_file=data/test.json
per_device_train_batch_size=1
per_device_eval_batch_size=1
gradient_accumulation_steps=8
max_seq_length=1024
output_dir="output"
exp_name=lora_model

lora_b_nums=3  # Developer-specific, k-means, or DBSCAN et al.
```

I evaluate the performance on MMLU with opencompass v0.2.0 and the following commands:
```bash
PYTHONPATH="/home/fine-tuning/HydraLoRA/HydraLoRA:$PYTHONPATH" python run.py --models hf_llama2_7b --datasets mmlu_ppl_ac766d 
```
I also commented out several lines in `opencompass/configs/datasets/mmlu/mmlu_ppl_ac766d.py`, following the same modifications mentioned in [issue 10](https://github.com/Clin0212/HydraLoRA/issues/10#issuecomment-2505449814), to avoid some errors. And I got a weighted average score of 40.12, still lower than the reported 47.22.  Have you got any new results?

**Comment by MrGH-boss** [comment]
@bhsimon0810  Hello, I hope you're doing well. Could you kindly share how you were eventually able to complete the evaluation? I'm encountering the same error that appeared in your screenshot and would really appreciate any guidance you could offer. Thank you very much!

**Comment by bhsimon0810** [comment]
@MrGH-boss Hi, you should set the `tokenizer` to the path of the saved `peft_path` (the tokenizer associated with the saved peft model), rather than the original pretrained model path. That should work. Feel free to let me know if you have further questions.
